Rename session events

spacebear21 opened the pull request:
_Follow up to discussion in https://github.com/payjoin/rust-payjoin/pull/1106#discussion_r2379691110. Note that I stuck with past tense over imperative mood because IMO that indicates more clearly that this event happen-ed (e.g. `RetrievedOriginalPayload` would be difficult to express in imperative mood), also we already use past tense for `Created` and `Closed`._

SessionEvent variants should describe "what event happened" instead of "what this event produces". This makes the inner values less ambiguous and better abides by the principle that events only capture primary information.

We already do this for the `Created` and `Closed` variants. `SessionInvalid` was left as-is because it will be replaced soon.

<details>
  <summary>Pull Request Checklist</summary>

Please confirm the following before requesting review:

- [x] I have [disclosed my use of
  AI](https://github.com/payjoin/rust-payjoin/blob/master/.github/CONTRIBUTING.md#ai-assistance-notice)
  in the body of this PR.
- [x] I have read [CONTRIBUTING.md](https://github.com/payjoin/rust-payjoin/blob/master/.github/CONTRIBUTING.md#commits) and **rebased my branch to produce [hygienic commits](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#committing-patches)**.
</details>

arminsabouri reviewed (approved):
cACK af66471bb1291462df2275a2cda1b9bb96c2c635

spacebear21 commented:
Changes since last review:

- I changed the names of the variants `CheckedBroadcastSuitability` and `CheckedNoInputsSeenBefore` to match the transition method names.
- The second commit updates the `apply_event` transition methods.

Note that `CommitedFees` doesn't match the method name `apply_fee_range`. This is intentional because I believe we should separate fee application from fee commitment similar to how we do in WantsOutputs/WantsInputs as discussed in https://github.com/payjoin/rust-payjoin/pull/1108.

DanGould reviewed (approved):
ACK 7586c537f571464fd26a243dab05e9ada8ce2ff6 LGTM
